c10/xpu/test/impl/XPUTest.cpp

```diff
@@ -0,0 +1,69 @@
+#include <gtest/gtest.h>
+
+#include <c10/xpu/XPUFunctions.h>
+
+#define ASSERT_EQ_XPU(X, Y) \
+  {                         \
+    bool _isEQ = X == Y;    \
+    ASSERT_TRUE(_isEQ);     \
+  }
+
+bool has_xpu() {
+  return c10::xpu::device_count() > 0;
+}
+
+TEST(XPUTest, DeviceCount) {
```

Comment:
Please rename the test suite name because it is too general.

Reply:
rename to XPUDeviceTest